### PR TITLE
Rename java package to data prepper

### PR DESCRIPTION
**Pull request by kowshikn**
*Description of changes:*

* Renamed the package from `.situp.` to `.dataprepper.`
* Renamed class names that uses `*Situp*` to `*DataPrepper*`
* Replaces usage of situp in logs, string constants. java docs and readme.

*Testing*

* Build was successful

```
./gradlew build

Deprecated Gradle features were used in this build, making it incompatible with Gradle 7.0.
Use '--warning-mode all' to show the individual deprecation warnings.
See https://docs.gradle.org/6.5/userguide/command_line_interface.html#sec:command_line_warnings

BUILD SUCCESSFUL in 2m 45s
75 actionable tasks: 60 executed, 15 up-to-date
```

* Tested the Jaeger and Our sample app by building a new image.

* Created the macos and linux tars.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: data-prepper-core/src/main/java/com/amazon/dataprepper/DataPrepperExecute.java [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+/**
+ * Execute entry into Data Prepper.
+ */
+public class DataPrepperExecute {
+    private static final Logger LOG = LoggerFactory.getLogger(DataPrepperExecute.class);
+
+    public static void main(String[] args) {
+        final DataPrepper dataPrepper = DataPrepper.getInstance();
+        if (args.length > 0) {
+            dataPrepper.execute(args[0]);
```
Comment (erosas) [inline data-prepper-core/src/main/java/com/amazon/dataprepper/DataPrepperExecute.java:15]: Why don't we pass all the parameters?

File: data-prepper-plugins/common/src/main/java/com/amazon/dataprepper/plugins/PluginRepository.java [hunk 2]
```diff
@@ -32,10 +32,10 @@
 
     private static void findPlugins() {
         final Reflections reflections = new Reflections(DEFAULT_PLUGINS_CLASSPATH);
-        final Set<Class<?>> annotatedClasses = reflections.getTypesAnnotatedWith(SitupPlugin.class);
+        final Set<Class<?>> annotatedClasses = reflections.getTypesAnnotatedWith(DataPrepperPlugin.class);
         for (final Class<?> annotatedClass : annotatedClasses) {
-            final SitupPlugin tiPluginAnnotation = annotatedClass
-                    .getAnnotation(SitupPlugin.class);
+            final DataPrepperPlugin tiPluginAnnotation = annotatedClass
```
Comment (erosas) [inline data-prepper-plugins/common/src/main/java/com/amazon/dataprepper/plugins/PluginRepository.java:37]: some old references to TI

File: data-prepper-core/src/main/java/com/amazon/dataprepper/DataPrepperExecute.java (hunk 1 above)
Comment (yadavcbala) [reply data-prepper-core/src/main/java/com/amazon/dataprepper/DataPrepperExecute.java:15]: Good point - When we added this we were only looking at having the config file name - but yeah I agree, we should have passed all args. This is out of scope of this PR so I would take it up as a separate issue and fix it in separate PR. @erosas  u ok with it ?

**Review by erosas — APPROVED** [review]
Found and old reference to TI in a variable, added a comment to mark it.
Also, why do we only pass the first argument to the runner?

File: data-prepper-plugins/common/src/main/java/com/amazon/dataprepper/plugins/PluginRepository.java (hunk 2 above)
Comment (kowshikn) [reply data-prepper-plugins/common/src/main/java/com/amazon/dataprepper/plugins/PluginRepository.java]: Fixed it.

**Review by yadavcbala — COMMENTED** [review]
Can you check the failing rawEndToEnd tests

**Review by yadavcbala — APPROVED** [review]
Approving - Discussed offline about the next commit for build issue

**Pull request merged**